### PR TITLE
Handle nulls properly in flatmap reader

### PR DESCRIPTION
**Pull request by zzhao0**
In the case that a file has very sparse non-null flatmap, reader incorrectly calculates totalMapEntries if the entire batch only contains nulls. That causes exception when reading map values from the child readers.

Test plan 

Adding a test file that contains all null flat maps except for the first row. This allows this particular edge case to be tested. Ie. the first batch returned contains 1024 rows, one of which is not null. The subsequent batches returned have only nulls.

== NO RELEASE NOTE ==

### CODE REVIEW

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatBatchStreamReader.java [hunk 1]
```diff
@@ -158,13 +158,7 @@ public Block readBlock()
         else {
             nullVector = new boolean[nextBatchSize];
             int nullValues = presentStream.getUnsetBits(nextBatchSize, nullVector);
-            if (nullValues == nextBatchSize) {
-                for (int i = 0; i < inMapStreams.size(); i++) {
-                    inMapVectors[i] = null;
-                    totalMapEntries += nextBatchSize;
```
Comment (arunthirupathi) [inline presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatBatchStreamReader.java:164]: Just for my understanding, I believe this is the real bug. But inMapVectors[i] is already null as it was just allocated in the top so this code is no-op,so this entire code is removed.

**Review by arunthirupathi — APPROVED** [review]
The change and the test looks good to me. Thanks for finding and fixing this bug.

File: presto-orc/src/test/java/com/facebook/presto/orc/TestMapFlatBatchStreamReader.java [hunk 2]
```diff
@@ -314,6 +315,17 @@ public void testWithAllNulls()
                 ExpectedValuesBuilder.get(Function.identity()).setNullRowsFrequency(ALL));
     }
 
+    @Test
+    public void testWithAllNullsExceptFirst()
+            throws Exception
+    {
+        // A test case where everyflat map is null except the first one
```
Comment (mbasmanova) [inline presto-orc/src/test/java/com/facebook/presto/orc/TestMapFlatBatchStreamReader.java:322]: typo: everyflat -> every flat

File: presto-orc/src/test/java/com/facebook/presto/orc/TestMapFlatSelectiveStreamReader.java [hunk 3]
```diff
@@ -313,6 +314,17 @@ public void testWithAllNulls()
                 ExpectedValuesBuilder.get(Function.identity()).setNullRowsFrequency(ALL));
     }
 
+    @Test
+    public void testWithAllNullsExceptFirst()
+            throws Exception
+    {
+        // A test case where everyflat map is null except the first one
```
Comment (mbasmanova) [inline presto-orc/src/test/java/com/facebook/presto/orc/TestMapFlatSelectiveStreamReader.java:321]: typo: everyflat -> every flat

File: presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatBatchStreamReader.java (hunk 1 above)
Comment (zzhao0) [reply presto-orc/src/main/java/com/facebook/presto/orc/reader/MapFlatBatchStreamReader.java:164]: exactly

**Comment by bhhari** [comment]
@zzhao0 you should also update MapFlatSelectiveStreamReader. Batch readers are going to be deprecated in future

**Comment by zzhao0** [comment]
@bhhari I added the test for MapFlatSelectiveStreamReader, it didn't fail

**Comment by zzhao0** [comment]
BTW, the failing test is in presto-mongodb module and doesn't seem relavent

For query: 
 SELECT count(*) FROM customer WHERE NOT EXISTS(SELECT * FROM orders WHERE orders.custkey=customer.custkey)
not equal
Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):
    [499]
Expected rows (up to 100 of 1 missing rows shown, 1 rows in total):
    [500]
	at org.testng.Assert.fail(Assert.java:94)
	at com.facebook.presto.tests.QueryAssertions.assertEqualsIgnoreOrder(QueryAssertions.java:223)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:203)
	at com.facebook.presto.tests.QueryAssertions.assertQuery(QueryAssertions.java:106)
	at com.facebook.presto.tests.AbstractTestQueryFramework.assertQuery(AbstractTestQueryFramework.java:140)
	at com.facebook.presto.tests.AbstractTestQueries.testCorrelatedExistsSubqueriesWithEqualityPredicatesInWhere(AbstractTestQueries.java:3735)

**Review by bhhari — APPROVED** [review]

**Pull request merged**